File: Chapter03/Exercise3_02/Shell.scala
package Exercise3_02



import org.apache.spark.sql.SparkSession

object Shell extends App {
  val spark: SparkSession = SparkSession.builder
    .master(s"local[4]")
    .appName("Shell")
    .getOrCreate()

  import Utilities02.HelperScala.{sampleWarcLoc, extractRawRecords, parseRawWarc}
  import Exercise2_06.Exercise2_06.heavyComputation

  val rawRecords = extractRawRecords(sampleWarcLoc)(spark)
  val warcRecords = rawRecords.flatMap(parseRawWarc)
  val invokeHeavyRDD = warcRecords.map(heavyComputation)








}

File: Chapter03/Exercise3_02/Shell.py
from pyspark.sql import SparkSession
from Chapter02.utilities02_py.helper_python import sample_warc_loc, extract_raw_records, parse_raw_warc
from Chapter02.Exercise2_06.Exercise2_06 import heavy_computation

spark: SparkSession = SparkSession.builder \
    .appName('SubmitWithMaster') \
    .getOrCreate()

raw_records = extract_raw_records(sample_warc_loc, spark)
warc_records = raw_records.flatMap(parse_raw_warc)
invoked_heavy_rdd = warc_records.map(lambda record: heavy_computation())
print(invoked_heavy_rdd.collect())